### ISSUE TITLE
Feature Suggestion: File size in app description

### ISSUE THREAD

**Issue by ghost** [Botspot/pi-apps#1339]
Hi all,

A common issue I have with Pi-Apps and some other app marketplaces is that they do not show the file size on the "about" or "description" page. This is quite an annoyance to me as my Pi only has a 16GB SD card, with it nearly having 3/4 of it used up. Since you can't see the file size until the application is finished installing, I would greatly appreciate the addition of #a file size indicator on the Details page.

Thank you in advance for any replies!

**Comment by Botspot** [comment]
With apt, dpkg, and Add/Remove Software, this is easy because deb files have a rigid standard. Deb files are basically a zip archive to contain everything necessary to install a program. To determine the installation size, you simply measure the total size of uncompressed files.

Compare that to Pi-Apps, where we use *scripts* to download files from the Internet and place them in various locations. There is no quick or easy way to measure installation size, because each script varies wildly in design.
I can only think of one way to accomplish that, and it would be to set up a dedicated Pi or VM to install each app and measure how full the sd card is after the operation. And that would still be very inaccurate:

- Many apps require the same dependency packages and share the same storage space. When measuring the installed size of an app, do we disregard the shared stuff? For example, many apps use openJDK. And it is around 150MB. When multiple apps are installed, they all share the same openJDK. How would that be accounted for?
- What about apps that consume a lot of storage when being run? Chromium, for example, fills up the `~/.config/chromium` folder with gigabytes of information over time. How would we account for that?

In other words, measuring the installation size for each app would be difficult, messy, inaccurate, and misleading. :(

**Comment by Crilum** [comment]
But, couldn't we measure the size of debs installed for package apps, and then put that in the info window?

**Comment by Botspot** [comment]
> But, couldn't we measure the size of debs installed for package apps, and then put that in the info window?

Currently we have 25 package-apps and 139 standard-apps. Sure, a size entry could be easily added for package-apps, but it would only work for the minority of apps that are package-apps.

**Comment by Serio0** [comment]
Thanks for the info!

**Comment by theofficialgman** [comment]
Closing as unactionable.
